app/controllers/users_controller.rb

```diff
@@ -1,7 +1,7 @@
 class UsersController < ApplicationController
-  before_action :set_user, only: [:edit, :update, :show, :destroy]
-  before_action :require_same_user, only: [:edit, :update, :destroy]
-  before_action :require_admin, only: [:destroy]
+  before_action :set_user, except: [:index, :new, :create]
+  before_action -> { require_same_user(@user, root_path) }, only: [:edit, :update]
```

Comment:
Any reason why you're using a Lambda here?

Reply:
The `require_same_user` method takes in 2 variables, to use it in a `before_action` I need to put it in a block